### PR TITLE
Stop testing 2.11 on nightly builds.

### LINKED ISSUE #134: Nightly GitHub Actions Build Fail on Sat, November 5th 2022
See run for more details:
https://github.com/TileDB-Inc/TileDB-CSharp/actions/runs/3398399411

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Nightly builds on 2.11 are failing because the `tiledb_status_code` function does not exist. This PR modifies `nightly.yml` to stop testing on 2.11.

Fixes #134 #135 #136 #140 #142. See [SC-23174](https://app.shortcut.com/tiledb-inc/story/23174/reconsider-testing-with-older-core-versions).

### CODE REVIEW

**Comment by shaunrd0** [comment]
LGTM pending approval from @ihnorton for dropping 2.11 from CI

**Review by ihnorton — APPROVED** [review]

**Pull request merged**
